sources/packages/backend/apps/api/src/route-controllers/note/_tests_/e2e/note.aest.controller.getInstitutionNotes.e2e-spec.ts

```diff
@@ -0,0 +1,190 @@
+    const institution = await institutionRepo.save(createFakeInstitution());
+    await saveFakeInstitutionNotes(
+      appDataSource,
+      [createFakeNote(NoteType.General)],
+      institution.id,
+    );
+    const endpoint = `/aest/note/institution/${institution.id}`;
+    const expectedPermissions = [
+      {
+        aestGroup: AESTGroups.BusinessAdministrators,
+        expectedHttpStatus: HttpStatus.OK,
+      },
+      {
+        aestGroup: AESTGroups.Operations,
+        expectedHttpStatus: HttpStatus.OK,
+      },
+      {
+        aestGroup: AESTGroups.OperationsAdministrators,
+        expectedHttpStatus: HttpStatus.OK,
+      },
+      {
+        aestGroup: AESTGroups.MOFOperations,
+        expectedHttpStatus: HttpStatus.OK,
+      },
+      {
+        aestGroup: undefined, // Read only user.
+        expectedHttpStatus: HttpStatus.OK,
+      },
+    ];
+
+    // Act/Assert
+    for (const permission of expectedPermissions) {
+      await request(app.getHttpServer())
+        .get(endpoint)
+        .auth(await getAESTToken(permission.aestGroup), BEARER_AUTH_TYPE)
+        .expect(permission.expectedHttpStatus);
+    }
+  });
+
+  it("Should get institution notes", async () => {
+    // Arrange
+    const institution = await institutionRepo.save(createFakeInstitution());
+    const [institutionNote] = await saveFakeInstitutionNotes(
+      appDataSource,
+      [createFakeNote(NoteType.General)],
+      institution.id,
+    );
+    const endpoint = `/aest/note/institution/${institution.id}`;
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .get(endpoint)
+      .auth(await getAESTToken(), BEARER_AUTH_TYPE)
+      .expect(HttpStatus.OK)
+      .then((response) => {
+        const [note] = response.body as NoteAPIOutDTO[];
+        expect(note.noteType).toBe(institutionNote.noteType);
+        expect(note.description).toBe(institutionNote.description);
+        expect(note.firstName).toBe(institutionNote.creator.firstName);
+        expect(note.lastName).toBe(institutionNote.creator.lastName);
+        expect(note.createdAt).toBe(institutionNote.createdAt.toISOString());
+      });
+  });
+
+  it("Should filter institution notes when a filter is provided", async () => {
+    // Arrange
+    const institution = await institutionRepo.save(createFakeInstitution());
+    await saveFakeInstitutionNotes(
+      appDataSource,
+      [
+        createFakeNote(NoteType.General),
+        createFakeNote(NoteType.Designation),
+        createFakeNote(NoteType.Program),
+      ],
+      institution.id,
+    );
+    const endpoint = `/aest/note/institution/${institution.id}?noteType=${NoteType.Designation}`;
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .get(endpoint)
+      .auth(await getAESTToken(), BEARER_AUTH_TYPE)
+      .expect(HttpStatus.OK)
+      .then((response) => {
+        const filteredNotes = response.body as NoteAPIOutDTO[];
+        for (const note of filteredNotes) {
```

Comment:
If only one is expected the `for` can be avoided and the length can also be checked.